src/main/java/com/blackducksoftware/integration/hub/api/component/ComponentRestService.java

```diff
@@ -48,4 +50,20 @@ public ComponentRestService(final RestConnection restConnection, final Gson gson
         return allComponents;
     }
 
+    public ComponentItem getExactComponentMatch(String id, String groupId, String artifactId, String version)
```

Comment:
Yes, id: is the criteria to lookup by. the entire string after the: is the 'id', and is has up to 4 parts:
- forge/namespace
- group
- artifact
- version

(for maven/gradle)

These can be different is a different forge/namespace is used. For instance, npm i believe has only 2 parts in most cases.